lib/methods/import.js

```diff
@@ -1,64 +1,437 @@
 const path = require('path');
 const fsp = require('node:fs/promises');
+const { cloneDeep } = require('lodash');
 const { EJSON } = require('bson');
 
 module.exports = self => {
   return {
-    async import(req, reporting) {
+    async import(req) {
+      const {
+        docs, attachmentsInfo, exportPath, filePath
+      } = await self.readExportFile(req);
       if (!req.user) {
         throw self.apos.error('forbidden');
       }
 
-      const { file } = req.files || {};
+      const total = docs.length + attachmentsInfo.length;
+      const {
+        reporting, jobId, notificationId
+      } = await self.instantiateJob(req, total);
 
-      if (!file) {
-        throw self.apos.error('invalid');
+      const {
+        duplicatedDocs, duplicatedIds, failedIds
+      } = await self.insertDocs(req, docs, reporting);
+      const importedAttachments = await self.insertAttachments(req, {
+        attachmentsInfo,
+        reporting,
+        duplicatedIds,
+        docIds: new Set(docs.map(({ aposDocId }) => aposDocId))
+      });
+
+      if (!duplicatedDocs.length) {
+        await reporting.end();
+
+        const notifMsg = `aposImportExport:${failedIds.length ? 'importFailedForSome' : 'importSucceed'}`;
+
+        await self.apos.notify(req, notifMsg, {
+          interpolate: {
+            count: failedIds.length
+          },
+          dismiss: true,
+          icon: 'database-import-icon',
+          type: failedIds.length ? 'danger' : 'success',
+          event: {
+            name: 'import-ended'
+          }
+        });
+        await self.apos.notification.dismiss(req, notificationId, 3000);
+
+        await self.cleanFile(exportPath);
+        await self.cleanFile(filePath);
+        return;
       }
 
-      // TODO: handle attachment files
-      const {
-        docs,
-        attachments
-        // attachmentsPath
-      } = await self.readExportFile(req, file);
+      // TODO: Should we stop the process of conflicts resolution
+      // if some docs encountered errors during import?
+      if (failedIds.length) {
+        self.apos.notify(req, 'aposImportExport:importFailedForSome', {
```

Comment:
Sometimes you `await self.apos.notify` and sometimes you don't.

Is it async?

Reply:
Added it back everywhere in the backend.